docs/reference/api/platform/Error.md

```diff
@@ -1,14 +1,123 @@
+### Error Status Usage
+Mbed OS pre-defines a set of system errors and **Error Status** is defined as a 32-bit Signed Integer in **NEGATIVE space**. The error status values are represented by **mbed_error_status_t** type.
 
+Error Status can be used  
+- To indicate return status from function calls
+  - Returning 0(=**MBED_SUCCESS**) or any positive number would be considered success.
+  - A predefined status code named **MBED_SUCCESS**(=0) is available to indicate success.
+
+- As system error codes
+  - Used to report error conditions to the OS error handling sub-system.
+
+### Error Status Types and Error Code Ranges
+Mbed OS error status encoding defines three types of error status values. They are as below. Note that type of error status is encoded into the 32-bit signed integer.
+- System Errors
+- Posix Errors
+- Custom Defined Errors
+
+And for each error status type, we have defined a range of error codes which forms the least significant 16-bits of 32-bit error status values. Based on error code definitions below are the error code ranges for each error type.
+- Posix Error Codes: 1 to 255 (see **Posix Error Codes Support** section below)
+- System Error Codes: 256 to 4095
+- Custom Error Codes: 4096 to 65535 
+
+Its important to understand the difference between the terminologies "**Error Code**" and "**Error Status**". "**Error Status**" is the 32-bit signed integer in **NEGATIVE space** used by applications to report the error, and "**Error Code**" is the least significant 16-bits of 32-bit Error Status values indicating the error situation(For example, Out of memory). All pre-defined error status definitions start with **MBED_ERROR_** prefix and all pre-defined error code definitions start with **MBED_ERROR_CODE_**.
+Error codes are not used by themselves, they are always encoded into error status values when reporting or representing errors in the system.
+
+Accordingly, the Error Status Ranges are as below for each type:
+- Posix Error Status  - 0xFFFFFFFF to 0xFFFFFF01 - This corresponds to Posix errors represented as negative.
+- System Error Status - 0x80XX0100 to 0x80XX0FFF - This corresponds to System errors range(all values are negative). Bits 23-16 will capture the module type (marked with XX)
+- Custom Error Status - 0xA0XX1000 to 0xA0XXFFFF - This corresponds to Custom errors range(all values are negative). Bits 23-16 will capture the module type (marked with XX)
+
+Mbed OS system error status values can also capture the module reporting the error by encoding the module identifier in the error status depending on the type of error status. The error status values/ranges are provided here for information purposes but no implementation should assume or manipulate the bit fields in the encoding directly. Implementation using Mbed OS error handling and reporting APIs should deal with error status like an opaque object and should use Mbed OS provided macros and functions to manipulate or interpret the error status values.
```

Comment:
Small nit: Should the last half of this segment (the part explaining that the bit fields do not need to be modified) be a bit higher up? This is a very important section. Putting a small blurb before diving in specifying that this section is closer to a technical reference though might help out new users a bit, it currently sounds like this needs to be manually configured at the bit field level until the end of the section.